### PR TITLE
flamenco: implement `add_new_reserved_account_keys` feature

### PR DESCRIPTION
**Pull request by mjain-jump**

### CODE REVIEW

File: src/flamenco/runtime/fd_system_ids.c [hunk 1]
```diff
@@ -35,46 +36,54 @@ const fd_pubkey_t fd_solana_spl_token_id                      = { .uc = { TOKEN_
...
+#define MAP_PERFECT_18      ( SYSVAR_STAKE_HIST_ID     ),
+#define MAP_PERFECT_19      ( NATIVE_LOADER_ID         ),
 
 #include "../../util/tmpl/fd_map_perfect.c"
+#undef PERFECT_HASH
 
 
 
-#define MAP_PERFECT_NAME fd_pubkey_builtin_prog_tbl
+/* https://github.com/firedancer-io/agave/blob/66c126b41ec2b55b3f747a4ac4e3ee6b439164a5/sdk/src/reserved_account_keys.rs#L152-L194 */
```
Comment (mjain-jump) [inline src/flamenco/runtime/fd_system_ids.c:82]: Move this comment above

File: contrib/test/txn-fixtures/program-tests.list [hunk 2]
```diff
@@ -28,3 +28,4 @@ dump/test-vectors/txn/fixtures/programs/deff84b60ea3c6284e621b4229f402c4274e4968
 dump/test-vectors/txn/fixtures/programs/ed4565e33252bca4dc606bdce4aa48a650e75048_1366919.fix
 dump/test-vectors/txn/fixtures/programs/crash-3354ca5c9ba1c2ea78c33855ec5ced47dcf9f29e.fix
 dump/test-vectors/txn/fixtures/programs/crash-6ac1f183ec533a9ef01c6b19ac8938ade3b12354.fix
+dump/test-vectors/txn/fixtures/programs/crash-38b53303426b0cb17570b27f89be658a890b867d.fix
```
Comment (mjain-jump) [inline contrib/test/txn-fixtures/program-tests.list:31]: Add another fixture for this same test case where `add_new_reserved_account_keys` is active

File: src/flamenco/runtime/fd_executor.c [hunk 3]
```diff
@@ -618,7 +618,7 @@ fd_executor_setup_accessed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
 int
 fd_should_set_exempt_rent_epoch_max( fd_rent_t const *       rent,
                                      fd_borrowed_account_t * rec ) {
-  if( fd_pubkey_is_sysvar_id( rec->pubkey ) ) {
+  if( !memcmp( rec->const_meta->info.owner, fd_sysvar_owner_id.key, sizeof(fd_pubkey_t) ) ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_executor.c:621]: can you add a permalink for this. Sometimes agave uses all sysvars and sometimes they reference a list used by this function

**Review by jumpsiegel — APPROVED** [review]

File: src/flamenco/runtime/fd_executor.c [hunk 4]
```diff
@@ -1117,10 +1089,9 @@ fd_executor_setup_borrowed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
 
       /* All new accounts should have their rent epoch set to ULONG_MAX. 
          https://github.com/firedancer-io/agave/blob/1e460f466da60a63c7308e267c053eec41dc1b1c/svm/src/account_loader.rs#L484-L494 */
-      if( FD_UNLIKELY( is_unknown_account ) ) {
+      if( is_unknown_account 
+          || ( i>0UL && fd_should_set_exempt_rent_epoch_max( &txn_ctx->epoch_ctx->epoch_bank.rent, borrowed_account ) ) ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_executor.c:1093]: messy

**Review by jumpsiegel — APPROVED** [review]

**Pull request merged**